File: ui/src/types/ApprovalWorkflow/eventConditionMaster.ts
import { array, boolean, Decoder, nullable, number, object, string } from "decoders";

export interface EventConditionMasterListDetail {
  MasterColumnId: number;
  MasterTableId: number;
  ApprovalEventId: number;
  ColumnName: string;
  ColumnDisplayName,
  ColumnSequence: number;
  ValueType:string;
  TableName: string;
  TableSequence: number
  }

  export const eventConditionMasterListDetailDecoder : Decoder<EventConditionMasterListDetail> = object({
  MasterColumnId: number,
  MasterTableId: number,
  ApprovalEventId: number,
  ColumnName: string,
  ColumnDisplayName:string,
  ColumnSequence: number,
  ValueType:string,
  TableName: string,
  TableSequence: number
  });  
  
  export interface EventConditionMasterList{
    EventConditionMasters:EventConditionMasterListDetail[]
  }

  export const eventConditionMasterListDecoder: Decoder<EventConditionMasterList> = object({
    EventConditionMasters: array(eventConditionMasterListDetailDecoder)
  });

File: api/BeSureApi/Models/EventConditionMaster.cs
namespace BeSureApi.Models
{
    public class EventConditionMasterColumnList
    {
        public int MasterColumnId { get; set; }
        public int MasterTableId { get; set; }
        public int ApprovalEventId { get; set; }
        public string ColumnName { get; set; }
        public string ColumnDisplayName { get; set; }
        public int ColumnSequence { get; set; }
        public string ValueType { get; set; }
        public string TableName { get; set; }
        public int TableSequence { get; set; }
    }
}
